### PR TITLE
network: introduce KeepMaster= setting

### LINKED ISSUE #20624: networkd: support [Network] Bridge=inherit
**Is your feature request related to a problem? Please describe.**

We operate clusters of `systemd-nspawn` containers created based on `mkosi` templates. We generate a per-container `.nspawn` that sets `Bridge=` to one of several bridges that already exist and also have physical Ethernet ports.

We also have `80-container-vb.network` the adds some settings to the host-side `veth` (e.g. `EmitLLDP=`). The very presence of this `.network` file matching our `vb-$machine` interfaces causes the interface to be removed from the bridge that `systemd-nspawn` just joined it to! This is regardless of the absence of a `Bridge=` option.

**Describe the solution you'd like**
```
[Network]
Bridge=inherit
```
> "inherit" will maintain any existing bridge membership.

**Describe alternatives you've considered**

Generating and removing `.network` files for every container is impractical due to volume and ephemeral nature of many containers especially considering the number of `networkctl reload`s required.

Tested `KeepConfiguration=static` - that might be another avenue here. Extend its meaning to be "don't remove any existing settings that aren't specifically being set" but that would be challenging to implement.

**The systemd version you checked that didn't have the feature you are asking for**
v247, v249

### PR DESCRIPTION
**Pull request by yuwata**
Closes #20624.

### CODE REVIEW

File: test/test-network/conf/23-keep-master.network [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[Match]
+Name=dummy98
+
+[Network]
+Bond=bond199
+ActiveSlave=true
```
Comment (bluca) [inline test/test-network/conf/23-keep-master.network:8]: did you intend to add KeepMaster=yes to this test file?

Comment (bluca) [reply test/test-network/conf/23-keep-master.network:8]: nvm, the test file was added in the previous commit but with fewer settings - maybe it should be moved to the next commit entirely?

**Review by bluca — APPROVED** [review]
lgtm

File: test/test-network/conf/23-keep-master.network (hunk 1 above)
Comment (yuwata) [reply test/test-network/conf/23-keep-master.network:8]: Ouch. Moved.

**Comment by yuwata** [comment]
@bluca Thank you for your review. Addressed the above comment. Upgrading the green label.

**Comment by bluca** [comment]
```
17:44:26 ======================================================================
17:44:26 ERROR: test_bond_keep_master (__main__.NetworkdBondTests)
17:44:26 ----------------------------------------------------------------------
17:44:26 Traceback (most recent call last):
17:44:26   File "./test/test-network/systemd-networkd-tests.py", line 3412, in test_bond_keep_master
17:44:26     copy_unit_to_networkd_unit_path('23-keep-master.network')
17:44:26   File "./test/test-network/systemd-networkd-tests.py", line 452, in copy_unit_to_networkd_unit_path
17:44:26     shutil.copy(os.path.join(networkd_ci_path, unit), network_unit_file_path)
17:44:26   File "/usr/lib64/python3.6/shutil.py", line 245, in copy
17:44:26     copyfile(src, dst, follow_symlinks=follow_symlinks)
17:44:26   File "/usr/lib64/python3.6/shutil.py", line 120, in copyfile
17:44:26     with open(src, 'rb') as fsrc:
17:44:26 FileNotFoundError: [Errno 2] No such file or directory: '/run/networkd-ci/23-keep-master.network'
17:44:26 
17:44:26 ======================================================================
17:44:26 ERROR: test_bridge_keep_master (__main__.NetworkdBridgeTests)
17:44:26 ----------------------------------------------------------------------
17:44:26 Traceback (most recent call last):
17:44:26   File "./test/test-network/systemd-networkd-tests.py", line 3590, in test_bridge_keep_master
17:44:26     copy_unit_to_networkd_unit_path('23-keep-master.network')
17:44:26   File "./test/test-network/systemd-networkd-tests.py", line 452, in copy_unit_to_networkd_unit_path
17:44:26     shutil.copy(os.path.join(networkd_ci_path, unit), network_unit_file_path)
17:44:26   File "/usr/lib64/python3.6/shutil.py", line 245, in copy
17:44:26     copyfile(src, dst, follow_symlinks=follow_symlinks)
17:44:26   File "/usr/lib64/python3.6/shutil.py", line 120, in copyfile
17:44:26     with open(src, 'rb') as fsrc:
17:44:26 FileNotFoundError: [Errno 2] No such file or directory: '/run/networkd-ci/23-keep-master.network'
```

**Comment by yuwata** [comment]
Ouch. I wrongly moved the conf file on the previous rebase. I hope it now fixed. Setting the green label again.

**Pull request merged**